examples/advanced/edge/README.md

```diff
@@ -67,34 +82,83 @@ You will then see the device start receiving the model from the server and compl
+we usually do not start with real devices. Therefore, NVFlare provides DeviceSimulator to give simulated devices for testing the FL process
 
 Let's run an end-to-end example with Cifar10 dataset with baseline comparisons.
+
 ### Baselines
+First, let's run the centralized baseline on the whole dataset, and a 16-client federated baseline via NVFlare's standard single-layer pipeline using [JobAPI](https://nvflare.readthedocs.io/en/main/programming_guide/fed_job_api.html).
 1. Run the centralized baseline
 ```commandline
 cd baselines
 python cifar_train_central.py
+cd ..
 ```
 2. Run the federated baseline under regular single-layer setting
 ```commandline
+cd baselines
 python cifar_fl_base_job.py
+cd ..
+```
+Let's visualize the results:
+```commandline
+tensorboard --logdir=/tmp/nvflare/workspaces
 ```
+With the centralized training of 10 epochs, and the federated training of 10 rounds (1 local epoch per round), you should see the following results:
+<img src="./figs/cifar10.png" alt="Cifar10 Results" width="800" >
+The two learning will converge to similar accuracy, note that in this case each client holds partial data that is 1/16 of the whole training set sequentially split.
+
+### Simulated Cross-Device Federated Learning
+Assuming the previous steps are completed, we can now run the end-to-end example with the same already prepared NVFlare system.
+#### Step1: Start the NVFlare System
+Again, we first start the system, open a new terminal window and run the following command:
+```commandline
+./start_nvflare.sh
+```  
+
+#### Step2: Start Local Web Proxy
+To route devices to its target, routing_proxy is used. It's a simple proxy that routes the request to
+the target based on checksum of the device ID. Open a new terminal and start the routing proxy:
+```commandline
+python ../../../nvflare/edge/web/routing_proxy.py 8000 /tmp/nvflare/workspaces/edge_example/prod_00/lcp_map.json
+```
+The lcp_map.json file is generated by tree_prov.py.
 
-## [Optional] Run Device Emulator
+#### Step3: Start Edge Simulation with DeviceSimulator
+The DeviceSimulator can be used to test all the features of the federated system. 
 
-If you don't have a device at hand or want to develop a new algorithm,
-you can utilize our NVFlare device emulator.
+To start the DeviceSimulator with a config, open a new terminal window, let's use the PyTorch version of the Cifar10 DeviceSimulator:
 
-The emulator can be used to test all the features of the edge system.
+```commandline
+export PYTHONPATH="${PYTHONPATH}:${PWD}/simulators"
+python ../../../nvflare/edge/device_simulator/run_device_simulator.py ./simulators/configs/cifar10_pt_config.json
+```
 
-To start the emulator with an emulator config:
+The DeviceSimulator polls the NVFlare system for job assignments. It runs one job and then quits.
+Please refer to [DeviceSimulator docs](../../../nvflare/edge/device_simulator/README.md) for more details.
 
+#### Step4: Start NVFlare Job
```

Comment:
these steps are duplicated of what's above.
if we want, maybe merge them, and explain they can do DeviceSimulator if they don't have a real iOS device?

Reply:
want to make this part self contained